### DISCUSSION TITLE
Tags based on Mentions example

### DISCUSSION THREAD

**Discussion by Deliaz** [ianstormtaylor/slate#4563] [Q&A] [answered]
Hi there, 

I am trying to adopt [Mentions example](https://www.slatejs.org/examples/mentions) but make it work with #tags instead. 
Almost succeed, except suggestion list. 

Currently, it shows the dropdown after inserting `#`-char **plus** any letter, however, I want to make it show suggestions right after the `#` being inserted. 
<img src="https://user-images.githubusercontent.com/4865430/135498002-9b15fa50-169c-406f-80d3-02a160927726.png" width="300"/>

Here is the code example: https://codesandbox.io/s/slate-tags-editor-ckkm1?file=/src/editor.tsx
I assume it requires some changes on lines 155-165 but still didn't figure it out. 

Thank you for any ideas!

**Comment by TheSpyder** [comment, marked as answer]
It's more complicated than it looks. The entire premise of that code is that there needs to be two words, it finds the point before them both and then checks if the string of the two words is `#<text>`.

Doing it for a single character means a dedicated search just for it:
https://codesandbox.io/s/slate-tags-editor-forked-1imot?file=/src/editor.tsx

**Reply by Deliaz** [reply]
Thank you! It works! 
Indeed, the original slate example misses an explanation of how it works. Which makes it a little bit hard to understand and modify
